merkletree/proof.go

```diff
@@ -36,6 +42,16 @@ func (n *ProofNode) hash(treeNonce []byte) []byte {
 	}
 }
 
+// AuthenticationPath is a pruned tree containing
+// the prefix path between the corresponding leaf node
+// (of type ProofNode) and the root. This is a proof
+// of inclusion or absence of requested index.
+// The proof type can be determined by
```

Comment:
Maybe just say, a proof of inclusion is when the leaf index equals the lookup index?